cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRunnerSystemTerminator.java

```diff
@@ -14,12 +14,14 @@
  * the License.
  */
 
-package io.cdap.cdap.k8s.common;
+package io.cdap.cdap.app.preview;
 
 /**
- * Enum representing Kubernetes resource types which can be run as twill applications on the Kubernetes cluster.
+ * Interface to terminate the preview runner system when no more preview runners are available.
  */
-public enum KubeResourceType {
-  DEPLOYMENT,
-  STATEFULSET
+public interface PreviewRunnerSystemTerminator {
```

Comment:
Shouldn't need this interface. In the `PreviewRunnerMain`, when adding the `PreviewRunnerManager` to the core service, add a listener such that on the Service completion, call `System.exit(0);`

Reply:
done.